platform/engineering/request-for-comment/tracking-submissions-from-origin-to-destination.md

```diff
@@ -0,0 +1,385 @@
+- Not be considered PII/PHI (but may use PII/PHI in its generation)
+
+## Design
+
+The main design is to generate a unique ID for each submission, regardless of type that can persist between all systems the submission will travel. In order to support systems that cannot or will not be updated to accept and work with this ID, it should be deterministic based on the content of the submission. This allows systems that are aware of the ID to "regenerate" the ID or corelate that system's ID to this unique ID.
+
+### Unique and Deterministic
+
+A globally unique ID must be generated for each submission. This ID should be deterministic such that the same submission should result in the same ID. This should be independent of time or place, allowing the ID to be generated or regenerated successfully at any point in the submission's lifecycle.
+
+### Format
+
+An ID should have the following qualities:
+
+- Consistent length
+- Resistent to collisions
+- Deterministic
+- Does not communicate data or qualities used to generate the ID
+
+Given these qualities, a [Hash Function](https://en.wikipedia.org/wiki/Hash_function) is a good fit.
+The size of the resulting hash should provide enough address space to handle a large number of submissions over the lifetime. I recommend a minimum of 256 bits to give ample room.
+
+Given this, I think the common [SHA2](https://en.wikipedia.org/wiki/SHA-2) hashing algorithm with a 256 bit digest will provide the best combination of speed, size, and common usage across multiple languages and services.
+
+### Content
+
+Now that we have criteria set, the question becomes What do we hash?
+
+#### Common Items
+
+Some items will be common between all submissions types and therefor always included in the hash.
+
+##### Submission Type
+
+Each type of submission should have a unique name, referenced in a global space. This could be a registry or a simple location where submission types could be declared/registered for common use.
+
+Defining these could be its own RFC but a basic description would be a list of Submission Types and their IDS in a tabular format:
+
+|    Submission Type     |           ID           |
+| :--------------------: | :--------------------: |
+|    VA Form 20-0995     |    va_form_20-0995     |
+|    VA Form 20-0996     |    va_form_20-0996     |
+| VA Evidence Submission | va_evidence_submission |
+|          etc           |          etc           |
+
+Ideally Submission Type IDs should be pre-normalized however this can also be handled during the normalization step of the ID generation.
+
+##### Submission Subject ID
+
+Submissions should be tied to a subject, be it a Veteran, Dependent, Representative, etc. The assumption is that all items tracked should be related to a singular subject. The submission should have some form of identifier as part of its content and we can use that identifier to both increase its uniqueness as well as deterministically identify it in a global space.
```

Comment:
With non-authenticated users, this could be difficult.

Reply:
Update - I read further and it seems like that is what you are thinking, but that might be confusing when trying to search for a veteran and not realizing that it was from the form body.

Reply:
Hopefully each subject type would have a SME that could define how and where to gather this data. Something like:
Form A uses field 2: Veterans Social Security Number
Form B uses field 3: VA File Number
...etc